### ISSUE TITLE
[BUG] C# Dev Kit does not support .NET9 on windows

### ISSUE THREAD

**Issue by Fluxionist** [microsoft/vscode-dotnettools#1633]
### Describe the Issue

The C# Dev Kit does not support .NET 9 on Windows.

### Steps To Reproduce

Steps:
1. Install C# Dev Kit 
2. .NET Install Tool is also installed
3. .NET Install Tool starts downloading .NET 8.  Manually stop it.
4. Set dotnetAcquisitionExtension.sharedExistingDotnetPath to .NET 9 folder.
5.  .NET Install Tool found .NET 9
6. Run command: .NET New Project...
7. Nothing happens
8. console- OUTPUT - C# Dev Kit   shows the following error message:
Starting Spawn .NET server...
IdeBenefitsSource: Failed to fetch entitlements. Error: 'Error: Request to https://api.subscriptions.visualstudio.microsoft.com/Me/Entitlements/IDEBenefits?api-version=2023-03-26&caller=vscode failed with status code: 401 and response ""'
Using local .NET runtime at "F:\dotnet\dotnet.exe"
Found installed dotnet location "F:\dotnet\dotnet.exe" to load projects.
Project system hosting process will use runtime installed in SDK.
.NET server STDERR: You must install or update .NET to run this application.

App: c:\Users\xxx\.vscode\extensions\ms-dotnettools.csdevkit-1.14.8-win32-x64\components\vs-green-server\platforms\win32-x64\node_modules\@microsoft\visualstudio-server.win32-x64\Microsoft.VisualStudio.Code.Server.exe
Architecture: x64

**.NET server STDERR: Framework: 'Microsoft.NETCore.App', version '8.0.0' (x64)**
.NET location: F:\dotnet

**The following frameworks were found:
  9.0.0 at [F:\dotnet\shared\Microsoft.NETCore.App]**

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=8.0.0&arch=x64&rid=win-x64&os=win10

.NET server exited with 2147516566

### Expected Behavior

".NET New Project..." command should open a dialog for new project

### Environment Information

Environment:
- Windows 11
- vscode 1.95.3
- .NET 9.0.0 (manually installed before this extension)
- C# Dev Kit 1.14.8 pre-release
- dotnet.exe in PATH

**Comment by webreidi** [comment]
Possibly a duplicate of https://github.com/microsoft/vscode-dotnettools/issues/1606?

**Comment by Michael-Eng** [comment]
This is not a dup of #1606. 

This bug is because our Ready2Run assemblies are built against .NET8 so we require that version of the runtime. We also have some APIs which have been deprecated in .NET9 and need to migrate.

**Comment by timheuer** [comment]
@Kobi-Blade This is not accurate. You can completely use VS Code for .NET 9 projects.  What is happening in this bug is the OP (@Fluxionist) stopped the installation of the .NET 8 **runtime** and expected things to still work.  There are components of the extension that require the NET8 runtime until we transition off of that. It's perfectly fine to have the NET8 runtime installed and also work on NET9 projects. Our team does this daily.

The bottom line is that there are some extension binaries that still have a dependency on NET8...but this does not affect the functionality of supporting building NET9 apps.

We'll close this as the OP is stopping the dependency required.  We're working on moving our dependency to NET9 as well.

**Comment by timheuer** [comment]
> working on moving dependencies to .NET 9, when we already starting to prepare for .NET 10 shows VS Code simply cannot keep up with .NET releases, in contrast to VS IDE

We're a bit behind yes in updating dependencies, but also like other dev teams evaluate any immediate needs. .NET 10 is 8 months away and we don't want our extension releases to carry non go-live bits as of right now. VS IDE doesn't depend on .NET 9 FYI.
